src/main/java/io/supertokens/usermetadata/UserMetadata.java

```diff
@@ -34,13 +34,8 @@ public static JsonObject updateUserMetadata(Main main, @Nonnull String userId, @
         return storage.startTransaction((con) -> {
             JsonObject originalMetadata = storage.getUserMetadata_Transaction(con, userId);
 
-            final JsonObject updatedMetadata = originalMetadata == null ? new JsonObject() : originalMetadata;
-            metadataUpdate.entrySet().forEach((entry) -> {
-                updatedMetadata.remove(entry.getKey());
-                if (!entry.getValue().isJsonNull()) {
-                    updatedMetadata.add(entry.getKey(), entry.getValue());
-                }
-            });
+            JsonObject updatedMetadata = originalMetadata == null ? new JsonObject() : originalMetadata;
+            MetadataUtils.shallowMergeMetadataUpdate(updatedMetadata, metadataUpdate);
```

Comment:
You may want to extrac this function to a global utils file. Since another recipe would be using it. So it would be weird if that recipe will refer directly to Metadata recipe.

Reply:
I've moved the entire `MetadataUtils` into the `Utils` folder. It's still global this way, but I felt like the `Utils` class was getting crowded.

Reply:
ok